filebeat/harvester/log.go

```diff
@@ -30,15 +35,15 @@ func (h *Harvester) Harvest() {
 	//       for new lines in input stream. Simple 8-bit based encodings, or plain
 	//       don't require 'complicated' logic.
 	cfg := h.Config
-	readerConfig := logFileReaderConfig{
-		forceClose:         cfg.ForceCloseFiles,
-		closeOlder:         cfg.CloseOlder,
-		backoffDuration:    cfg.Backoff,
-		maxBackoffDuration: cfg.MaxBackoff,
-		backoffFactor:      cfg.BackoffFactor,
+	readerConfig := reader.LogFileReaderConfig{
+		ForceClose:         cfg.ForceCloseFiles,
+		CloseOlder:         cfg.CloseOlder,
+		BackoffDuration:    cfg.Backoff,
+		MaxBackoffDuration: cfg.MaxBackoff,
+		BackoffFactor:      cfg.BackoffFactor,
 	}
 
-	reader, err := createLineReader(
+	rd, err := reader.CreateLineReader(
```

Comment:
I'm a little bit undecided here TBH because I think parts could also be used for a binary file harvester for example. I suggest to keep it in a separate package for now and move it to the right place as soon as we have more harvester / reader types?